lab2+scanner/grammar/Grammar.py

```diff
@@ -19,6 +20,7 @@ def read(self):
             self.N = [value.strip() for value in file.readline().strip().split(',')]
             self.E = [value.strip() for value in file.readline().strip().split(',')]
             self.S = file.readline().strip()
+            self.file_S = self.S
             rules = [value.strip() for value in ''.join([line for line in file]).strip().split(',')]
             self.i = 1
 
```

Comment:
on line 30 you can use also a list for lhs because there can be multiple nonTerminals